mod_form.php

```diff
@@ -292,4 +300,24 @@ public function add_completion_rules() {
     public function completion_rule_enabled($data) {
         return true;
     }
+    public function validation($data, $files) {
+        $errors = parent::validation($data, $files);
+
+        if (!empty($data['emailperiod']) && $data['emailperiod'] <= 60 && $data['emailfrequency'] > 5) {
+            // Email is sent based on minutes selector - make sure no more than 5 emails can be sent.
+            $errors['emailfrequency'] = get_string('frequencytoohigh', 'reengagement', 5);
```

Comment:
I was thinking about the use-case of something "time-sensitive" - eg - a quiz that they must complete by a certain time. - but I think you're right - we should make them more restrictive.